src/components/Common/DateTimeInput.tsx

```diff
@@ -0,0 +1,39 @@
+import { format } from "date-fns";
+import * as React from "react";
+
+import { Input } from "@/components/ui/input";
+
+function toISOWithTimezone(localVal: string): string {
+  const localDate = new Date(localVal);
+  return localDate.toISOString();
+}
+
+function toLocalDateTimeString(isoString: string): string {
+  const date = new Date(isoString);
+  return format(date, "yyyy-MM-dd'T'HH:mm");
+}
+
+type DateTimeInputProps = {
+  value?: string;
+  onDateChange?: (val: string) => void;
+} & React.InputHTMLAttributes<HTMLInputElement>;
```

Comment:
i'd say:
1. let's extend `Input` instead of html's `input`, since in the component's code we are actually doing that.
2. avoid overriding `value`, it'll be inherited from `Input`'s `value` prop.

```suggestion
type DateTimeInputProps = React.ComponentProps<typeof Input>{
  onDateChange?: (val: string) => void;
};
```